internal/rbd/rbd_util.go

```diff
@@ -566,6 +566,7 @@ func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 			depth++
 		}
 		vol.RbdImageName = vol.ParentName
+		vol.Pool = vol.ParentPool
```

Comment:
Same issue as before, re-using a rbdVolume with different attributes is quite wrong. Needs cleanup urgently.